### PR TITLE
[clang][cas] Prefix map -fdepfile-entry

### PR DESCRIPTION
**Pull request by benlangmuir**
For rdar://106307646

### CODE REVIEW

**Comment by benlangmuir** [comment]
@swift-ci please test

**Comment by akyrtzi** [comment]
It's not clear to me why this is necessary, in `createCompileJobCacheKeyImpl()` these are not included in the key:
```
  // These are added in when the dependency file is generated, but they don't
  // affect the actual compilation.
  DepOpts.ExtraDeps.clear();
```

**Comment by benlangmuir** [comment]
It shows up in the module context hash.

File: clang/lib/Tooling/DependencyScanning/ScanAndUpdateArgs.cpp [hunk 1]
```diff
@@ -171,6 +171,10 @@ void DepscanPrefixMapping::remapInvocationPaths(CompilerInvocation &Invocation,
   Mapper.mapInPlace(CodeGenOpts.ProfileInstrumentUsePath);
   Mapper.mapInPlace(CodeGenOpts.SampleProfileFile);
   Mapper.mapInPlace(CodeGenOpts.ProfileRemappingFile);
+
+  // Dependency output options.
+  for (auto &ExtraDep : Invocation.getDependencyOutputOpts().ExtraDeps)
```
Comment (akyrtzi) [inline clang/lib/Tooling/DependencyScanning/ScanAndUpdateArgs.cpp:186]: Could you add a comment that this is a workaround? "Proper" handling for these may be that we should treat these options as "transparent" (not affecting caching) and pass them through unchanged.

**Review by akyrtzi — APPROVED** [review]
Some additional comments would be helpful, otherwise LGTM.

**Comment by benlangmuir** [comment]
@swift-ci please test

**Pull request merged**